pkg/aws/eni/node.go

```diff
@@ -452,3 +452,16 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 
 	return available, nil
 }
+
+// GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
+// that can be allocated to the instance
+func (n *Node) GetMaximumAllocatableIPv4(scopedLog *logrus.Entry) int {
+	limits, limitsAvailable := n.getLimits()
+	if !limitsAvailable {
+		scopedLog.Warning("Could not determined instance limits, maximum allocatable ipv4 addresses will be 0 (unlimited)" +
+			" this could lead to allocation overflows if the max-allocate flag is not set")
+		return 0
+	}
+
+	return (limits.Adapters - *n.k8sObj.Spec.ENI.FirstInterfaceIndex) * limits.IPv4
```

Comment:
Is it guaranteed that `limits.Adapters` > 0?

Reply:
indeed, I revised the logic to handle this case 👍